meson.build

```diff
@@ -0,0 +1,144 @@
+# -*- mode: meson -*-
+
+# Style objective: be consistent with what mesonbuild.com documents/uses, and/or
+# the meson book: https://meson-manual.com/
+
+project(
+  'i3lock',
+  'c',
+  version: '2.13',
+  default_options: [
+    'c_std=c11',
+    'warning_level=1',  # enable all warnings (-Wall)
+    # TODO(https://github.com/i3/i3/issues/4087): switch to
+    # 'buildtype=debugoptimized',
+  ],
+  # Ubuntu 18.04 (supported until 2023) has meson 0.45.
+  # We can revisit our minimum supported meson version
+  # if it turns out to be too hard to maintain.
+  meson_version: '>=0.45.0',
+)
+
+cc = meson.get_compiler('c')
+add_project_arguments(cc.get_supported_arguments(['-Wunused-value']), language: 'c')
+
+if meson.version().version_compare('>=0.48.0')
+  # https://github.com/mesonbuild/meson/issues/2166#issuecomment-629696911
+  meson.add_dist_script('meson/meson-dist-script')
+else
+  message('meson <0.48.0 detected, dist tarballs will not be filtered')
+endif
+
+################################################################################
+# Version handling
+################################################################################
+
+cdata = configuration_data()
+
+version_array = meson.project_version().split('.')
+cdata.set_quoted('I3LOCK_VERSION', '@VCS_TAG@')
+cdata.set_quoted('SYSCONFDIR', join_paths(get_option('prefix'), get_option('sysconfdir')))
+
+if get_option('b_sanitize').split(',').contains('address')
+  cdata.set('I3LOCK_ASAN_ENABLED', 1)
+endif
+
+cdata.set('HAVE_STRNDUP', cc.has_function('strndup'))
+cdata.set('HAVE_MKDIRP', cc.has_function('mkdirp'))
+
+# Instead of generating config.h directly, make vcs_tag generate it so that
+# @VCS_TAG@ is replaced.
+config_h_in = configure_file(
+  output: 'config.h.in',
+  configuration: cdata,
+)
+config_h = declare_dependency(
+  sources: vcs_tag(
+    input: config_h_in,
+    output: 'config.h',
+    fallback: meson.project_version() + '-non-git',
+  )
+)
+
+################################################################################
+# manpages
+################################################################################
+
+man1 = join_paths(get_option('mandir'), 'man1')
+
+install_data(
+  [
+	'i3lock.1',
+  ],
+  install_dir: man1,
+)
```

Comment:
What's wrong with the `install_man('i3lock.1')` function?

Reply:
Thanks, done in https://github.com/i3/i3lock/pull/320